openshift/aws-vpce-operator
# Recreate Rejected VPC Endpoints

This commit handles the case when a VPC Endpoint is put into the rejected state by deleting the VPC Endpoint. This has the net effect of the controller recreating the VPC Endpoint.

Test plan in stage:
* Create a stage HCP cluster
* Reject it's VPC Endpoint
* Observe the updated code delete the VPC Endpoint, recreate it, and update Route53

It takes quite some time for it to get to a steady state (~30 minutes), so there's room for improvement as we're polling AWS every 15minutes by default if we don't run into errors, but this helps the uninstall case where we don't even need to wait for the VPC Endpoint to get into an available state.

Fixes [OSD-17219](https://issues.redhat.com//browse/OSD-17219)
